script/generate_missing_latest_version_views

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+"""Generate one view definition file per document type in '_stable' tables.
+
+Run as:
+  ./script/generate_missing_latest_version_views 'moz-fx-data-shared-prod:*_stable.*'
+"""
+
+from argparse import ArgumentParser
+from google.cloud import bigquery
+from typing import Callable, Iterable, Tuple
+from fnmatch import fnmatchcase
+import logging
+import os
+import re
+import sys
+
+VERSION_RE = re.compile(r"_v([0-9]+)$")
+WHITESPACE_RE = re.compile(r"\s+")
+WILDCARD_RE = re.compile(r"[*?[]")
+
+DEFAULT_PATTERN = "telemetry.*"
+DEFAULT_EXCLUDE = r"*_raw"
+
+client = bigquery.Client()
+
+def pattern(value: str) -> Tuple[str, str, str]:
+    PATTERN_RE.fullmatch(value).groups(client.project)
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument(
+    "patterns",
+    metavar="[project:]dataset[.table]",
+    default=[DEFAULT_PATTERN],
+    nargs="*",
+    help="Table that should have a latest-version view, may use shell-style wildcards,"
+    f" defaults to: {DEFAULT_PATTERN}",
+)
+parser.add_argument(
+    "--exclude",
+    action="append",
+    default=[DEFAULT_EXCLUDE],
+    metavar="project:dataset.table",
+    help="Latest-version views that should be ignored, may use shell-style wildcards,"
+    f" defaults to: {DEFAULT_EXCLUDE}",
+)
+parser.add_argument(
+    "--log-level",
+    default="INFO",
+    help="Defaults to INFO",
+)
+
+args = parser.parse_args()
+
+# set log level
+try:
+    logging.basicConfig(level=args.log_level, format="%(levelname)s %(message)s")
+except ValueError as e:
+    parser.error(f"argument --log-level: {e}")
+
+wildcards = set("*?[]")
+def uses_wildcards(pattern: str) -> bool:
+    return bool(set("*?[]") & set(pattern))
+
+patterns = [
+    (project, dataset, table)
+    for pattern in args.patterns
+    for project, _, dataset_table in (pattern.partition(":"),)
+    for dataset, _, table in (dataset_table.partition("."),)
+]
+all_projects = None
+all_datasets = {}
+all_tables = {}
+views = {}
+for pattern in args.patterns:
+    project, _, dataset_table = pattern.partition(":")
+    dataset, _, table = dataset_table.partition(".")
+    projects = [project or client.project]
+    dataset = dataset or "*"
+    table = table or "*"
+    if uses_wildcards(project):
+        if all_projects is None:
+            all_projects = {p.project_id for p in client.list_projects()}
+        projects = [p for p in all_projects if fnmatchcase(project, p)]
+    for project in projects:
+        datasets = [dataset]
+        if uses_wildcards(dataset):
+            if project not in all_datasets:
+                all_datasets[project] = {
+                    d.dataset_id for d in client.list_datasets(project)
+                }
```

Comment:
this should probably be a list comprehension, not a set comprehension

Reply:
nevermind, i remember now why not

Reply:
i take that back, there's not reason for this, or `all_projects` to be set comprehensions instead of list comprehensions